### DISCUSSION TITLE
Forge server Connaction refused

### DISCUSSION THREAD

**Discussion by SpyTheEngineer** [itzg/docker-minecraft-server#1680] [Q&A]
So, im trying to make a 1.19 forge server so i can screw around with my friends, I have my own server computer thats running ubuntu server and the command line i ran to install the modded server is this

docker run -d -it --name moddedmc -v /home/spy/moddedmc:/data -e SERVER_PORT=25564 -e VERSION=1.19 -e TYPE=FORGE -e EULA=TRUE -e MEMORY=8G itzg/minecraft-server

but when i try to connect to it using forge and the mod i installed, the connection just says refused, what am I doing wrong and how do i fix it? any and all help is appreciated

**Comment by itzg** [comment]
If you're wanting to run the server on port 25564 (guessing from the variable you set), then replace `-e SERVER_PORT=25564` with a port mapping like `-p 25564:25565`.

**Reply by SpyTheEngineer** [reply]
So i need a -p? I did not know that, ill try and get back to you.

**Reply by itzg** [reply]
Correct. Inside the container, it listens to port 25565*, but the cool thing about containers is that can port-map the inside port to whatever outside port you want, such as 25564 in this case.

\* Technically you can change that listening port inside the container with `SERVER_PORT`; however, that tends to complicate other things.

**Reply by SpyTheEngineer** [reply]
I reinstalled it, with the -p 25564:25565 and removed the SERVER_PORT, it did not work, I do have both 25565 and 25564 port forwarded, but 25565 is for my vanilla server.

**Reply by SpyTheEngineer** [reply]
I got it working! Thank you so much for your help! You are amazing, and this entire page is so well documented! With your help i was able to finally figure the problem out.
